src/gotcha.c

```diff
@@ -161,7 +178,9 @@ GOTCHA_EXPORT enum gotcha_error_t gotcha_wrap(struct gotcha_binding_t* user_bind
 
   for (lib_iter = _r_debug.r_map; lib_iter != 0; lib_iter = lib_iter->l_next) {
     debug_printf(2, "Looking for wrapped callsites in %s\n", LIB_NAME(lib_iter));
-    FOR_EACH_PLTREL(lib_iter, gotcha_wrap_impl, lib_iter, bindings, num_actions);
+    if(libraryFilterFunc(lib_iter)){
```

Comment:
I assume the desired semantics are that someone could write a set of MPI wrappings, then use the filter function to limit the search to libmpi.so.

But this is the code that looks through callsites in every library for calls to libmpi.so.  If we filter here then we'll have filtered-callsites going to libmpi.so and unfiltered-callsites going to the tool.  I'm not imagining any cases where that's what the tool writer wants.

The libraryFilter could safely limit our gotcha_prepare_symbols step (not that that's a huge performance problem area).  Or it could filter the the callsites by only looking at libraries that depend on libmpi.so.  But I don't think it should be filtering all libraries.